core/src/main/scala/kafka/server/ReplicaFetcherThread.scala

```diff
@@ -135,10 +97,10 @@ class ReplicaFetcherThread(name: String,
       // to avoid failing the caller, especially during shutdown. We will attempt to close
       // leaderEndpoint after the thread terminates.
       try {
-        leaderEndpoint.initiateClose()
+        leader.initiateClose()
       } catch {
         case t: Throwable =>
-          error(s"Failed to initiate shutdown of leader endpoint $leaderEndpoint after initiating replica fetcher thread shutdown", t)
+          error(s"Failed to initiate shutdown of ReplicaFetcherBlockingSend after initiating replica fetcher thread shutdown", t)
```

Comment:
Could the log message print the leader endpoint instead (just like earlier from LHS)? You could always override the `toString()` inside the `leader` object to be able to print a human readable version of the object here.